satpy/readers/sar_c_safe.py

```diff
@@ -438,29 +437,29 @@ def expand(self, chunks):
 
     @property
     def first_pixel(self):
-        return int(self.element.find("firstRangeSample").text)
+        return np.uint16(self.element.find("firstRangeSample").text)
 
     @property
     def last_pixel(self):
-        return int(self.element.find("lastRangeSample").text)
+        return np.uint16(self.element.find("lastRangeSample").text)
 
     @property
     def first_line(self):
-        return int(self.element.find("firstAzimuthLine").text)
+        return np.uint16(self.element.find("firstAzimuthLine").text)
 
     @property
     def last_line(self):
-        return int(self.element.find("lastAzimuthLine").text)
+        return np.uint16(self.element.find("lastAzimuthLine").text)
 
     @property
     def lines(self):
         lines = self.element.find("line").text.split()
-        return np.array(lines).astype(int)
+        return np.array(lines).astype(np.uint16)
 
     @property
     def lut(self):
         lut = self.element.find("noiseAzimuthLut").text.split()
-        return np.array(lut).astype(float)
+        return np.array(lut).astype(np.float32)
```

Comment:
Would it be more performant in this way so that the intermediate array wouldn't be created?
```suggestion
        return np.array(lut, dtype=np.float32)
```

Reply:
I think you could also pass `copy=False` to `astype`.

Reply:
I like @pnuu's solution